tests/codebase/test_code_quality.py

```diff
@@ -130,7 +130,7 @@ def test_this_file(fname: str, test_file: IO[str]) -> None:
         if path.startswith(exclude_dirs):
             continue
 
-        with open(path, "r", encoding="utf-8") as file:
+        with open(path, "r", encoding="utf-8", newline="") as file:
```

Comment:
Not sure there is any cross-platform solution here short of opening the file in binary mode and searching. But I think I'd propose just getting rid of this check altogether before doing that, given it has not really been an issue in practice (as evidenced by the small number of files needing update in this PR).

Reply:
`newline` is supposed to make it cross platform and it actually works. I tested this on Windows and it fails as expected without the second commit and succeeds with it. I'm guessing this failure may have something to do with repository checkout or settings in CI.

Reply:
in that case, if the test works for local dev use, we could simply go back to running the codebase checks on a single platform, and still get the correct errors reported. I don't actually see what the benefit of running on all three is.